### PR TITLE
`fixed_regex_linter()` doesn't fail with `"\\/"`

### LINKED ISSUE #1845: `fixed_regex_linter()` fails with "unrecognized escape" error
A minimal version of code I found in the wild that seems to cause problems for `fixed_regex_linter()`:

``` r
strsplit("a/b", "\\/")
#> [[1]]
#> [1] "a" "b"

library(lintr)

lint(
  text = 'strsplit("a/b", "\\/")',
  linters = fixed_regex_linter()
)
#> <text>:1:18: error: [error] '\/' is an unrecognized escape in character string starting ""\/"
#> strsplit("a/b", "\/")
#>                  ^
```

<sup>Created on 2022-12-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #1845

### CODE REVIEW

File: R/fixed_regex_linter.R [hunk 1]
```diff
@@ -231,7 +235,7 @@ get_token_replacement <- function(token_content, token_type) {
       token_content
     }
   } else { # char_escape token
-    if (rex::re_matches(token_content, rex::rex("\\", one_of("^${}().*+?|[]\\<>:;")))) {
+    if (rex::re_matches(token_content, rex::rex("\\", one_of("^${}().*+?|[]\\<>:;/")))) {
```
Comment (MichaelChirico) [inline R/fixed_regex_linter.R:238]: would you mind checking the ASCII table for any other common symbols that might be worth including?

Comment (IndrajeetPatil) [reply R/fixed_regex_linter.R]: Good idea. I have expanded on the character set. I think this about covers all characters we might encounter in this context.

File: tests/testthat/test-fixed_regex_linter.R [hunk 2]
```diff
@@ -40,6 +40,7 @@ test_that("fixed_regex_linter blocks simple disallowed usages", {
   expect_lint("regexec('\\\\$', x)", lint_msg, linter)
   expect_lint("grep('\n', x)", lint_msg, linter)
   expect_lint("grep('\\\\;', x)", lint_msg, linter)
+  expect_lint("grep('\\\\/', x)", lint_msg, linter)
```
Comment (MichaelChirico) [inline tests/testthat/test-fixed_regex_linter.R:43]: maybe we need a patrick test for all the characters, wdyt?

Comment (IndrajeetPatil) [reply tests/testthat/test-fixed_regex_linter.R:43]: SGTM. Will add them.

Comment (IndrajeetPatil) [reply tests/testthat/test-fixed_regex_linter.R]: I have added parametrized tests now.

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**